staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -3831,10 +3835,24 @@ type PodStatus struct {
 	// +optional
 	NominatedNodeName string `json:"nominatedNodeName,omitempty" protobuf:"bytes,11,opt,name=nominatedNodeName"`
 
-	// IP address of the host to which the pod is assigned. Empty if not yet scheduled.
+	// hostIP holds the IP address of the host to which the pod is assigned.
+	// Empty if the pod has not started yet.
+	// A pod can be assigned to a node that has a problem in kubelet which in turns mean that HostIP will
+	// not be updated even if there is node is assigned to pod
 	// +optional
 	HostIP string `json:"hostIP,omitempty" protobuf:"bytes,5,opt,name=hostIP"`
-	// IP address allocated to the pod. Routable at least within the cluster.
+
+	// hostIPs holds the IP addresses allocated to the host. If this field is specified, the first entry must
+	// match the hostIP field. This list is empty if the pod has not started yet.
+	// A pod can be assigned to a node that has a problem in kubelet which in turns means that HostIPs will
+	// not be updated even if there is a node is assigned to this pod.
+	// +optional
+	// +patchStrategy=merge
+	// +patchMergeKey=ip
+	// +listType=set
+	HostIPs []HostIP `json:"hostIPs,omitempty" protobuf:"bytes,14,rep,name=hostIPs" patchStrategy:"merge" patchMergeKey:"ip"`
```

Comment:
weirdly, [upgrade tests](https://testgrid.k8s.io/google-gce-upgrade#gce-stable1-latest-upgrade-master-parallel&width=20&include-filter-by-regex=Apply) started failing yesterday on this as well, with this error:

```
Kubernetes e2e suite: [sig-api-machinery] ServerSideApply should work for subresources
{ Failure /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/apimachinery/apply.go:162
Mar 29 23:11:48.373: Failed to Apply Status using Apply patch: failed to create manager for existing fields: failed to convert new object (apply-3390/test-pod; /v1, Kind=Pod) to smd typed: .status.hostIPs: element 0: associative list without keys has an element that's a map type
```

![Screen Shot 2022-03-30 at 11 31 28 PM](https://user-images.githubusercontent.com/980082/160970925-f0995f83-bfd4-4dc6-bd29-7d62559cc8ff.png)

Reply:
@apelisse - any idea what this could be about?

Reply:
For set, `patchStrategy` and `patchMergeKey` are not needed.
```
	// +patchStrategy=merge
	// +patchMergeKey=ip
```

Reply:
https://github.com/kubernetes/kubernetes/blob/ea0764452222146c47ec826977f49d7001b0ea8c/staging/src/k8s.io/api/core/v1/types.go#L2245-L2265

Reply:
Yeah, hostIp is a struct with one field, so it's tempting to use it in a set, but a set has to be made of scalar elements. You can use a `listType=map` with `listMapKey=ip` if you want to merge the IPs as unique items.

Reply:
I'm curious how the test passed in presubmit on this branch, but failed in the skew ... any ideas on that?

Reply:
Can it be a random change in the configuration that causes that node to use HostIPs when another test server doesn't?

Reply:
This PR is why I was asking for docs on these tags, @apelisse  :)